src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -0,0 +1,330 @@
+        }
+
+        private List<String> getSplitPathList(String path) {
+            String trimmedPath = path.replace("gs://", "");
+            return Lists.newArrayList(trimmedPath.split("/"));
+        }
+
+        protected String getBucketName(String path) {
+            List<String> splitPathList = getSplitPathList(path);
+            return splitPathList.remove(0);
+        }
+
+        // Get the path to the source file minus the scheme and bucket name
+        protected String getBlobName(String path) {
+            List<String> splitPathList = getSplitPathList(path);
+            // Remove the bucket name from the path
+            List<String> splitPathListNoBucket = splitPathList.subList(1, splitPathList.size());
+            return String.join(File.separator, splitPathListNoBucket);
+        }
+
+        public Set<String> schemesHandled() {
+            return new HashSet<>(Lists.newArrayList("gs"));
+        }
+
+        public boolean downloadFrom(String sourcePath, Path destination) {
+            Storage gsClient = getGoogleGSClient();
+
+            String bucketName = getBucketName(sourcePath);
+            String blobName = getBlobName(sourcePath);
+
+            //Blob blobMetadata = gsClient.get(bucketName, blobName, Storage.BlobGetOption.fields(Storage.BlobField.values()));
+            Blob blobMetadata;
+            try {
+                blobMetadata = gsClient.get(bucketName, blobName, Storage.BlobGetOption.fields(Storage.BlobField.values()));
+            } catch (StorageException e) {
+                System.err.println(
+                        "Could not get metadata for download source file " + sourcePath + " Storage exception:" + e.getMessage());
+                return false;
+            }
+
+            if (blobMetadata == null) {
+                System.err.println("Could not get metadata for download source file " + sourcePath + " Please check the file path.");
+                return false;
+            }
+
+            long inputSize = blobMetadata.getSize();
+
+            Blob blob;
+            try {
+                blob = gsClient.get(BlobId.of(bucketName, blobName));
+            } catch (StorageException e) {
+                System.err.println("Could not get info for dowload source file " + sourcePath + " Storage exception:" + e.getMessage());
+                return false;
+            }
+
+            // The destination path does not exist when a WDL workflow
+            // or tool is run so make sure the directories in the path exist
+            try {
+                Files.createDirectories(destination.getParent());
+            } catch (IOException e) {
+                System.err.println(
+                        "Could not create download destination file " + destination.toString() + " IO exception:" + e.getMessage());
+                return false;
+            }
+
+            ProgressPrinter printer = new ProgressPrinter();
```

Comment:
Don't think these (printer, fileContent, writeTo) get closed on exception
Note that a single try ... catch with resources can handle multiple resources so that might make things easier/cleaner

Reply:
done for fileContent and writeTo but printer doesn't have a close method and does not support auto-close?